system-test/speech.js

```diff
@@ -0,0 +1,411 @@
+      speech.createDetectJob(AUDIO_FILES.bridge.gcsUri, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, operation, apiResponse) {
+        assert.ifError(err);
+        assert(operation.name);
+        assert(apiResponse.name);
+        setTimeout(function() {
+          operation.get(function(err, operation, apiResponse) {
+            assert.ifError(err);
+            assert(Array.isArray(operation.response.results));
+            assert(operation.response.results.length > 0);
+            assert(operation.response.results[0].alternatives[0].transcript);
+            assert(operation.response.results[0].alternatives[0].confidence);
+            assert(apiResponse.response);
+            assert(apiResponse.response.typeUrl);
+            assert(apiResponse.response.value);
+            done();
+          });
+        }, 5000);
+      });
+    });
+    it('recognizes speech from remote audio file', function(done) {
+      speech.createDetectJob(AUDIO_FILES.bridge.httpUri, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, operation, apiResponse) {
+        assert.ifError(err);
+        assert(operation.name);
+        assert(apiResponse.name);
+        setTimeout(function() {
+          operation.get(function(err, operation, apiResponse) {
+            assert.ifError(err);
+            assert(Array.isArray(operation.response.results));
+            assert(operation.response.results.length > 0);
+            assert(operation.response.results[0].alternatives[0].transcript);
+            assert(operation.response.results[0].alternatives[0].confidence);
+            assert(apiResponse.response);
+            assert(apiResponse.response.typeUrl);
+            assert(apiResponse.response.value);
+            done();
+          });
+        }, 5000);
+      });
+    });
+  });
+
+  describe('createRecognizeStream', function() {
+    it('recognizes speech from raw audio', function(done) {
+      var through = require('through2');
+      var toRecognizeRequest = through.obj(function(chunk, encoding, done) {
+        done(null, {
+          audioContent: chunk
+        });
+      });
+
+      var recognizeStream = speech.createRecognizeStream({
+        streamingConfig: {
+          config: {
+            encoding: 'LINEAR16',
+            sampleRate: 16000
+          },
+          interimResults: false,
+          singleUtterance: false
+        }
+      });
+
+      recognizeStream.on('response', function(chunk) {
+        assert(chunk);
+        assert(chunk.code === 200);
+      });
+
+      var count = 0;
+
+      function checkIsDone() {
+        count++;
+        if (count === 3) {
+          done();
+        }
+      }
+
+      recognizeStream.on('data', function(chunk) {
+        var expected = 'how old is the Brooklyn Bridge';
+        assert(chunk);
+        if (chunk.endpointerType === 'START_OF_SPEECH') {
```

Comment:
WDYT about having some kind enum like structure for the end pointer types? I think this would be a useful feature for our users.